packages/apollo-codegen-core/src/loading.ts

```diff
@@ -197,14 +92,19 @@ export function extractDocumentFromJavascript(
 ): string | null {
   let tagName = options.tagName || 'gql';
 
-  const re = new RegExp(tagName + 's*`([^`]*)`', 'g');
   let match;
-  let matches = [];
+  let matches: string[] = [];
 
-  while ((match = re.exec(content))) {
-    const doc = match[1].replace(/\${[^}]*}/g, '');
+  try {
+    matches = extractDocumentsWithAST(content, options);
+  } catch (e) {
+    console.log('Extraction using AST failed with \n', e, '\nRetrying using regex');
+    const re = new RegExp(tagName + 's*`([^`]*)`', 'g');
```

Comment:
Removing the regex fallback makes sense. It will cause some configuration pain in some cases., i.e. actively excluding `.d.ts` files, since there are many that don't parse well. For the most part, I trust the parsers. Running it on spectrum and the engine frontend yielded good results.